gson/src/main/resources/META-INF/proguard/gson.pro

```diff
@@ -65,6 +65,6 @@
 # See also https://github.com/google/gson/pull/2420#discussion_r1241813541 for a more detailed explanation
 -if class *
```

Comment:
I think this `if class *`, `-classeswithmembers... class <1>` doesn't do anything relative to `-classeswithmembers... class *` , after reading some of the follow-up threads on that change.

Reply:
At least for R8 it does seem to make a difference: `-keepclasseswithmembers ... class *` always keeps the class, even if it isn't used. Whereas `-if class *  ... -keepclasseswithmembers ... class <1>` only keeps the class if it is actually used.

Unfortunately this is not covered by the tests yet; I will try to add a new test for that.

Reply:
Thanks also for the other points you raised, but unless sgjesse wants to address them here as well could you please create a separate GitHub issue or pull request? Otherwise the discussion on this pull request here might drift too far away from the original topic.